app/notifications/validators.py

```diff
@@ -57,31 +57,43 @@ def check_service_over_api_rate_limit(service, key_type):
                 raise RateLimitError(rate_limit, interval, key_type)
 
 
-def check_service_over_daily_message_limit(service, key_type):
+def check_service_over_daily_message_limit(service, key_type, notification_type=None):
     if key_type == KEY_TYPE_TEST or not current_app.config["REDIS_ENABLED"]:
         return 0
 
-    cache_key = daily_limit_cache_key(service.id)
+    limits = {
+        None: service.message_limit,
+        LETTER_TYPE: service.letter_message_limit,
+        EMAIL_TYPE: service.email_message_limit,
+        SMS_TYPE: service.sms_message_limit,
+    }
+    limit = limits[notification_type]
+    limit_group = notification_type or "total"
+
+    cache_key = daily_limit_cache_key(service.id, notification_type=notification_type)
     service_stats = redis_store.get(cache_key)
     if service_stats is None:
         # first message of the day, set the cache to 0 and the expiry to 24 hours
         service_stats = 0
         redis_store.set(cache_key, service_stats, ex=86400)
         return service_stats
-    if int(service_stats) >= service.message_limit:
+    if int(service_stats) >= limit:
         current_app.logger.info(
-            "service {} has been rate limited for daily use sent {} limit {}".format(
-                service.id, int(service_stats), service.message_limit
+            "service {} has been rate limited for daily use ({}): sent {}, limit {}".format(
+                service.id, limit_group, int(service_stats), limit
             )
         )
-        raise TooManyRequestsError(service.message_limit)
+        raise TooManyRequestsError(f"{limit_group}: {limit}")
     return int(service_stats)
 
 
-def check_rate_limiting(service, api_key):
+def check_rate_limiting(service, api_key, notification_type=None):
```

Comment:
i'd rather we didnt add this code yet if it's currrently unused (since we haven't updated how this function is called by post_notifications.py, and we haven't updated how `check_service_over_daily_message_limit` is called in tasks.py either)